python/foglamp/common/web/ssl_wrapper.py

```diff
@@ -0,0 +1,228 @@
+     -issuer_hash    - print issuer hash value
+     -issuer_hash_old    - print old-style (MD5) issuer hash value
+     -hash           - synonym for -subject_hash
+     -subject        - print subject DN
+     -issuer         - print issuer DN
+     -email          - print email address(es)
+     -startdate      - notBefore field
+     -enddate        - notAfter field
+     -purpose        - print out certificate purposes
+     -dates          - both Before and After dates
+     -modulus        - print the RSA key modulus
+     -pubkey         - output the public key
+     -fingerprint    - print the certificate fingerprint
+     -alias          - output certificate alias
+"""
+
+
+def get_x509(cmd):
+    if user_cert is None:
+        raise OSError("No user certificate supplied.")
+    echo_process = subprocess.Popen(['echo', user_cert], stdout=subprocess.PIPE)
+    a = subprocess.Popen(["openssl", "x509", "-noout", "-nameopt", "sep_comma_plus_space", cmd], stdin=echo_process.stdout, stderr=subprocess.STDOUT,)
+    outs, errs = a.communicate()
+    if outs is None and errs is None:
+        raise OSError(
+            'Error in executing command "{}"'.format("openssl x509 -noout {}".format(cmd)))
+    if errs is not None:
+        raise OSError(
+            'Error in executing command "{}". Error: {}, return code: {}'.format(cmd, errs.decode('utf-8').replace('\n', ''), a.returncode))
+    d = [b for b in outs.decode('utf-8').split('\n') if b != '']
+    return d
+
+
+def get_serial():
+    cmd = '-serial'
+    serial = get_x509(cmd=cmd)[0].split("serial=")[1].strip()
+    return serial
+
+
+def get_purposes():
+    cmd = '-purpose'
+    purposes = get_x509(cmd=cmd)
+    return purposes
+
+
+def get_issuer_common_name():
+    cmd = '-issuer'
+    issuer = get_x509(cmd=cmd)[0].split("issuer=")[1].strip()
+    return issuer
+
+
+def get_subject():
+    cmd = '-subject'
+    subject_text = get_x509(cmd=cmd)[0].split("subject=")[1].strip()
+    subject = subject_text.strip().split(", ")
+    subject_dict = {
+        "country": next(filter(lambda x: x.startswith("C="), subject), None).split("C=")[1],
+        "state": next(filter(lambda x: x.startswith("ST="), subject), None).split("ST=")[1],
+        "organisation": next(filter(lambda x: x.startswith("O="), subject), None).split("O=")[1],
+        "commonName": next(filter(lambda x: x.startswith("CN="), subject), None).split("CN=")[1],
+        "email": next(filter(lambda x: x.startswith("emailAddress="), subject), None).split("emailAddress=")[1],
+    }
+    return subject_dict
+
+
+def get_fingerprint():
+    cmd = '-fingerprint'
+    fp = get_x509(cmd=cmd)[0].split("SHA1 Fingerprint=")[1].strip()
+    return fp
+
+
+def get_pubkey():
+    cmd = '-pubkey'
+    pk = get_x509(cmd=cmd)[0].strip()
+    return pk
+
+
+def get_startdate():
+    cmd = '-startdate'
+    stdt = get_x509(cmd=cmd)[0].split("notBefore=")[1].strip()
+    return stdt
+
+
+def get_enddate():
```

Comment:
please also add a new `def is_expired` with return True / False, wrapper method of get_enddate